### PR TITLE
Project Editor Prefs

### PR DESCRIPTION
**Pull request by kkjamie**
Allows you to setup editor prefs that you can control via a tool window. These editor prefs can be used for anything editor time or editor runtime, such as drawing gizmos, skipping cutscenes etc...

EG: 
```
private void OnDrawGizmos()
{
	if (ProjectEditorPrefs.Get<bool>("drawMyFancyGizmos"))
	{
		Gizmos.color = Color.cyan;
		Gizmos.DrawCube(transform.position, new Vector3(3, 2, 3));
	}
}
```
image

### CODE REVIEW

**Comment by kkjamie** [comment]
Still requires documentation.

Also not 100% sure about the namespace, I could switch it to `ProjectEditorPrefs`

File: Prefs/ProjectEditorPrefs.cs [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+			return null;
+		}
+
+		private static void SetEditorPref(string key, object value, Type type)
+		{
+			if (type == typeof(string))
+			{
+				EditorPrefs.SetString(key, (string)value);
+			}
+			if (type == typeof(bool))
```
Comment (Rollo-Liu) [inline Prefs/ProjectEditorPrefs.cs]: `else if`?

File: Editor/ProjectEditorPrefsWindow.cs [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+		private void OnSelectionChange()
+		{
+			Repaint();
+		}
+
+		private void OnGUI()
+		{
+			var types = ProjectEditorPrefs.StoredPrefs;
+			foreach (var key in types.Keys.ToList())
```
Comment (Rollo-Liu) [inline Editor/ProjectEditorPrefsWindow.cs]: I would prefer to use `pair` iterate the dictionary.

```
foreach (var pair in ProjectEditorPrefs.StoredPrefs)
{
	var key = pair.Key;
	var type = pair.Value;
	var value = ProjectEditorPrefs.Get(key);
				
	......
}
```

File: Prefs/ProjectEditorPrefs.cs [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+using UnityEditor;
+using UnityEngine;
+
+#if UNITY_EDITOR
+namespace DUCK.Prefs
+{
+	public static class ProjectEditorPrefs
+	{
+		public static Dictionary<string, Type> StoredPrefs { get; private set; }
+		private readonly static string projectName;
```
Comment (Rollo-Liu) [inline Prefs/ProjectEditorPrefs.cs]: `private static readonly` -- Rider says. :trollface:

**Comment by kkjamie** [comment]
I have updated with the requested changes

**Pull request merged**